modules/jfx.incubator.richtext/src/main/java/jfx/incubator/scene/control/richtext/LineNumberDecorator.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2023, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jfx.incubator.scene.control.richtext;
+
+import java.text.DecimalFormat;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Label;
+
+/**
+ * Side decorator which shows paragraph (line) numbers.
+ * The numbering starts at line 1.
+ */
+public class LineNumberDecorator implements SideDecorator {
+    private final DecimalFormat format;
+
+    /**
+     * Creates an instance using Western-style group separator (comma).
+     */
+    public LineNumberDecorator() {
+        this("#,##0");
+    }
+
+    /**
+     * Creates an instance using the specified pattern for {@link DecimalFormat}.
+     * @param pattern the DecimalFormat pattern to use
+     */
+    public LineNumberDecorator(String pattern) {
```

Comment:
I think it would be more straight-forward to define `LineNumberDecorator(DecimalFormat format)` instead of taking a `String` pattern.

I recommend adding a getter for the format/pattern.